transformer_engine/common/comm_gemm_overlap/comm_gemm_overlap.cpp

```diff
@@ -21,12 +21,18 @@
 #define HALF_BYTES 2
 #define UB_MAX_SM 32
 
-#define AS_VECTOR(shape) std::vector<size_t>(shape.data, shape.data + shape.ndim)
```

Comment:
If we pass a function invocation here, this macro resolves to:
```
std::vector<size_t>(shape().data, shape().data + shape().ndim);
```
Each of those function returns will have a different location in the stack, so the constructor will do something weird. I was seeing errors from 18-D tensors.